src/Symfony/Component/Lock/Store/PdoStore.php

```diff
@@ -332,34 +332,17 @@ private function getDriver(): string
         } else {
             $driver = $con->getDriver();
 
-            switch (true) {
-                case $driver instanceof \Doctrine\DBAL\Driver\Mysqli\Driver:
-                    throw new \LogicException(sprintf('The adapter "%s" does not support the mysqli driver, use pdo_mysql instead.', static::class));
-                case $driver instanceof \Doctrine\DBAL\Driver\AbstractMySQLDriver:
-                    $this->driver = 'mysql';
-                    break;
-                case $driver instanceof \Doctrine\DBAL\Driver\PDOSqlite\Driver:
-                case $driver instanceof \Doctrine\DBAL\Driver\PDO\SQLite\Driver:
-                    $this->driver = 'sqlite';
-                    break;
-                case $driver instanceof \Doctrine\DBAL\Driver\PDOPgSql\Driver:
-                case $driver instanceof \Doctrine\DBAL\Driver\PDO\PgSQL\Driver:
-                    $this->driver = 'pgsql';
-                    break;
-                case $driver instanceof \Doctrine\DBAL\Driver\OCI8\Driver:
-                case $driver instanceof \Doctrine\DBAL\Driver\PDOOracle\Driver:
-                case $driver instanceof \Doctrine\DBAL\Driver\PDO\OCI\Driver:
-                    $this->driver = 'oci';
-                    break;
-                case $driver instanceof \Doctrine\DBAL\Driver\SQLSrv\Driver:
-                case $driver instanceof \Doctrine\DBAL\Driver\PDOSqlsrv\Driver:
-                case $driver instanceof \Doctrine\DBAL\Driver\PDO\SQLSrv\Driver:
-                    $this->driver = 'sqlsrv';
-                    break;
-                default:
-                    $this->driver = \get_class($driver);
-                    break;
+            if ($driver instanceof \Doctrine\DBAL\Driver\Mysqli\Driver) {
+                throw new \LogicException(sprintf('The adapter "%s" does not support the mysqli driver, use pdo_mysql instead.', static::class));
```

Comment:
why is it unsupported ? That looks suspicious to me. We already have the logic in place to use SQL compatible with MySQL, and I don't think we go low-level enough in our usage of DBAL APIs to be concerned about other kind of differences between drivers than the platform (which requires escaping the abstraction layer)

Reply:
for reference https://github.com/symfony/symfony/pull/35375